modin/dataframe/io.py

```diff
@@ -263,54 +357,56 @@ def read_csv(filepath_or_buffer,
+    elif not isinstance(filepath_or_buffer, pathlib.Path) and \
+            not isinstance(filepath_or_buffer, py.path.local):
+        warnings.warn(("Reading from buffer. "
+                       "Defaulting to Pandas implementation."),
+                      PendingDeprecationWarning)
+
+        return _read_csv_from_pandas(filepath_or_buffer, kwargs)
 
-        warnings.warn("Defaulting to Pandas implementation",
+    if _infer_compression(filepath_or_buffer, compression) is not None:
+        warnings.warn(("Compression detected. "
+                       "Defaulting to Pandas implementation."),
                       PendingDeprecationWarning)
 
-        pandas_obj = pandas.read_csv(filepath_or_buffer, **kwargs)
-        if isinstance(pandas_obj, pandas.DataFrame):
-            return from_pandas(pandas_obj, get_npartitions())
+        return _read_csv_from_pandas(filepath_or_buffer, kwargs)
 
-        return pandas_obj
+    if as_recarray:
+        warnings.warn("Defaulting to Pandas implementation.",
+                      PendingDeprecationWarning)
 
-    filepath = filepath_or_buffer
+        return _read_csv_from_pandas(filepath_or_buffer, kwargs)
 
-    # TODO: handle case where header is a list of lines
-    first_line = _get_firstline(filepath)
-    columns = _infer_column(first_line, kwargs=kwargs)
-    if header is None or (header == "infer" and names is not None):
-        first_line = b""
-        ignore_first_line = False
-    else:
-        ignore_first_line = True
-
-    offsets = _compute_offset(filepath, get_npartitions(),
-                              ignore_first_line=ignore_first_line)
-
-    # Serialize objects to speed up later use in remote tasks
-    first_line_id = ray.put(first_line)
-    kwargs_id = ray.put(kwargs)
-
-    df_obj_ids = []
-    index_obj_ids = []
-    for start, end in offsets:
-        if start != 0:
-            df, index = _read_csv_with_offset._submit(
-                args=(filepath, start, end, kwargs_id, first_line_id),
-                num_return_vals=2)
-        else:
-            df, index = _read_csv_with_offset._submit(
-                args=(filepath, start, end, kwargs_id),
-                num_return_vals=2)
-        df_obj_ids.append(df)
-        index_obj_ids.append(index)
+    if chunksize is not None:
+        warnings.warn(("Reading chunks from a file. "
+                       "Defaulting to Pandas implementation."),
+                      PendingDeprecationWarning)
+
+        return _read_csv_from_pandas(filepath_or_buffer, kwargs)
+
+    if skiprows is not None and not isinstance(skiprows, int):
+        warnings.warn(("Defaulting to Pandas implementation. To speed up "
+                       "read_csv through the Pandas on Ray implementation, "
+                       "comment the rows to skip instead."))
+
+        return _read_csv_from_pandas(filepath_or_buffer, kwargs)
+
+    # TODO: replace this by reading lines from file.
+    if nrows is not None:
```

Comment:
Alternate solutions to handling `nrows`:
1. Read file line-by-line. Requires more code, but `read_csv` will still be similar speed to Pandas because it requires reading all rows sequentially. Slight optimization when `nrows` is small.
2. Use current read_csv method and drop unneeded rows. Much slower than Pandas when `nrows` is small and the file is large, but could be faster when `nrows` is large and the file is large.